src/azure-cli/azure/cli/command_modules/servicefabric/custom.py

```diff
@@ -549,62 +552,68 @@ def remove_cluster_node(cmd, client, resource_group_name, cluster_name, node_typ
         raise CLIError("Failed to find the node type in the cluster")
 
     node_type = node_types[0]
-    if node_type.durability_level.lower() == 'bronze':
-        raise CLIError("Can't delete node if durability level is bronze")
-    reliability_level = _get_target_instance(cluster.reliability_level)
-    vmss = compute_client.virtual_machine_scale_sets.get(
-        resource_group_name, node_type.name)
+    reliability_required_instance_count = _get_target_instance(cluster.reliability_level)
+    vmss = _get_cluster_vmss_by_node_type(compute_client, resource_group_name, cluster.cluster_id, node_type.name)
     vmss.sku.capacity = vmss.sku.capacity - number_of_nodes_to_remove
-    if node_type.is_primary:
-        if vmss.sku.capacity < reliability_level:
-            raise CLIError("Can't delete node since current reliability level is {}".format(
-                reliability_level))
+    if vmss.sku.capacity < reliability_required_instance_count:
+        raise CLIError("Can't delete node since current reliability level is {} requires at least {} nodes.".format(
+            cluster.reliability_level,
+            reliability_required_instance_count))
+
+    # update vmss
     vmss_poll = compute_client.virtual_machine_scale_sets.create_or_update(
         resource_group_name, vmss.name, vmss)
     LongRunningOperation(cli_ctx)(vmss_poll)
-    node_type.vm_instance_count = vmss.sku.capacity
 
+    # update cluster
+    node_type.vm_instance_count = vmss.sku.capacity
     patch_request = ClusterUpdateParameters(node_types=cluster.node_types)
+
     return client.update(resource_group_name, cluster_name, patch_request)
 
 
 def update_cluster_durability(cmd, client, resource_group_name, cluster_name, node_type, durability_level):
     cli_ctx = cmd.cli_ctx
+
+    # get cluster node type durablity
     cluster = client.get(resource_group_name, cluster_name)
-    node_types = [n for n in cluster.node_types if n.name.lower() == node_type.lower()]
-    if node_types is None:
-        raise CLIError("Failed to find the node type in the cluster")
+    node_type_refs = [n for n in cluster.node_types if n.name.lower() == node_type.lower()]
+    if node_type_refs is None:
```

Comment:
It seems that `node_type_refs` will always be an empty list ([]) instead of None. We may use `if not node_type_refs:` here.

Reply:
fixed, thanks